### ISSUE TITLE
Imporving notebook cell output in a new view

### ISSUE THREAD

**Issue by LeonDpl** [jupyterlab/jupyterlab#4580]
Hi all,
Wondering if there is some work on following features:

- Outputing cell in a new view with cell inline command, with kind of built-in magic commands for instance. (Maybe it could be related to issue #4021)
- When a cell output is in a new view, hide/don't show the output in the source notebook (show output only in the new view)

Would appreciate to know if there is some similar work planned.
BR

**Comment by saulshanabrook** [comment]
> Outputing cell in a new view with cell inline command, with kind of built-in magic commands for instance. (Maybe it could be related to issue #4021)

Yeah, I think #4021 does cover it. With the added JS output support in https://github.com/jupyterlab/jupyterlab/pull/4515, it might be possible to access the juptyerlab API through the JS that is outputted by a cell and execute arbitrary commands. Although, at the moment I don't thinks that's recommended

. > When a cell output is in a new view, hide/don't show the output in the source notebook (show output only in the new view)

Does just collapsing the output work for this?

**Comment by LeonDpl** [comment]
>  Although, at the moment I don't thinks that's recommended

Should I understand that I should wait for similar behaviour or is it not planned at all ?

> Does just collapsing the output work for this?  

Thought that this could do the job but it seems that collapsed output is not persistent with cell execution (i.e. if an output is collapsed when I execute the related input the output is revealed). Is there already a linked issue ?

Many thnks

**Comment by saulshanabrook** [comment]
> Should I understand that I should wait for similar behaviour or is it not planned at all ?

Yeah, it is not planned in core at the moment, although it is certainly possible to create an extension that achieves this. I opened https://github.com/jupyterlab/jupyterlab/issues/4623 with more explanation. 

> Thought that this could do the job but it seems that collapsed output is not persistent with cell execution (i.e. if an output is collapsed when I execute the related input the output is revealed). Is there already a linked issue ? (found #3981 but execution problem is not mentionned)

I think that is a bug. I also thought there was an existing issue for it, but I can't find it at the moment.

**Comment by saulshanabrook** [comment]
Also see https://github.com/jupyterlab/jupyterlab/issues/3692 for more information on the first request.

**Comment by saulshanabrook** [comment]
> Thought that this could do the job but it seems that collapsed output is not persistent with cell execution (i.e. if an output is collapsed when I execute the related input the output is revealed).

@LeonDpl I changed the issue title to reflect this, since I believe the other point you brought up is already covered.
